client/data/marketplace/site-search-api.ts

```diff
@@ -0,0 +1,96 @@
+import { flatten } from 'q-flat';
+import { encode } from 'qss';
+import { RETURNABLE_FIELDS } from './constants';
+import type { SearchParams } from './types';
+
+// Maps sort values to values expected by the API
+const SORT_QUERY_MAP = new Map( [
+	[ 'oldest', 'date_asc' ],
+	[ 'newest', 'date_desc' ],
+	[ 'relevance', 'score_default' ],
+] );
+/**
+ * Map sort values to ones compatible with the API.
+ *
+ * @param {string} sort - Sort value.
+ * @returns {string} Mapped sort value.
+ */
+function mapSortToApiValue( sort: string ) {
+	// Some sorts don't need to be mapped
+	if ( [ 'price_asc', 'price_desc', 'rating_desc' ].includes( sort ) ) {
+		return sort;
+	}
+
+	return SORT_QUERY_MAP.get( sort ) ?? 'score_default';
+}
+
+/**
+ * Generate the query string for an API request
+ *
+ * @returns {string} The generated query string.
+ */
+function generateApiQueryString( { query, groupId, pageHandle, pageSize, locale }: SearchParams ) {
+	const sort = 'score_default';
+
+	let params: {
+		fields: string[];
+		filter?: { bool: { must: object[] } };
+		page_handle?: string;
+		query: string;
+		sort: string;
+		size: number;
+		group_id?: string;
+		from?: number;
+		lang: string;
+	} = {
+		fields: [ ...RETURNABLE_FIELDS ],
+		// filter: buildFilterObject( filter ),
+		page_handle: pageHandle,
+		query: encodeURIComponent( query ?? '' ),
+		sort: mapSortToApiValue( sort ),
+		size: pageSize,
+		lang: locale,
+	};
+
+	if ( groupId ) {
+		params = {
+			...params,
+			group_id: groupId,
+		};
+	}
+
+	// TODO implement author search
+	// if ( author ) {
+	// }
+
+	return encode( flatten( params ) );
```

Comment:
Instead of using two functions (and their npm packages) to encode and flatten the params, could we format them ourselves and avoid using functions?

Reply:
Agreed, I will modify that

Reply:
I am using the exiting `qs` package here and not adding any new dependency, but please let me know if you would prefer to use a function for this instead of that library.

Reply:
I have removed them from the `package.json`, are you reviewing the [latest commit](https://github.com/Automattic/wp-calypso/pull/63243/commits/e65e08a4a1d425fccb6592cd8e31e08e8fe448b9)?

Reply:
Aaa, yes thats great! Thanks @epeicher !